hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -1725,6 +1739,189 @@ public void delete(final Path path, final boolean recursive,
     } while (shouldContinue);
   }
 
+  /**
+   * Handles deletion of a path over Blob Endpoint.
+   * @param path path to be deleted
+   * @param recursive defines deletion to be recursive or not
+   * @param tracingContext object to trace the API flows
+   *
+   * @throws IOException exception from server call or if recursive used on
+   * non-empty directory
+   */
+  private void deleteBlobPath(final Path path,
+      final boolean recursive,
+      final TracingContext tracingContext) throws IOException {
+    StringBuilder listSrcBuilder = new StringBuilder();
+    String srcPathStr = path.toUri().getPath();
+    listSrcBuilder.append(srcPathStr);
+    if (!path.isRoot()) {
+      listSrcBuilder.append(FORWARD_SLASH);
+    }
+
+    String listSrc = listSrcBuilder.toString();
+    BlobList blobList = client.getListBlobs(null, listSrc, null, null,
+        tracingContext).getResult().getBlobList();
+    if (blobList.getBlobPropertyList().size() > 0) {
+      orchestrateBlobDirDeletion(path, recursive, listSrc, blobList, tracingContext);
+    } else {
+      LOG.debug(String.format("Path %s doesn't have child-blobs", srcPathStr));
+      if (!path.isRoot()) {
+        try {
+          LOG.debug(String.format("Deleting path %s", srcPathStr));
+          client.deleteBlobPath(path, null, tracingContext);
+          LOG.debug(String.format("Path deleted %s", srcPathStr));
+        } catch (AbfsRestOperationException ex) {
+          if (ex.getStatusCode() == HTTP_NOT_FOUND) {
+            LOG.error(String.format("Path %s doesn't exist", srcPathStr));
+            throw new AbfsRestOperationException(
+                ex.getStatusCode(),
+                AzureServiceErrorCode.PATH_NOT_FOUND.getErrorCode(),
+                ex.getErrorMessage(), ex);
+          }
+          throw ex;
+        }
+      }
+    }
+    createParentDirectory(path, tracingContext);
```

Comment:
Add comment that we are here only if src path was a non-empty directory, marker blob or normal blob, this is needed for parent to not go orphan which is taken care in wasb driver also.

Reply:
Added the comment now:
```
 /*
     * Src Path here would be either a non-empty directory, marker-blob or a
     * normal-blob. Path can not be non-existing at this point.
     * If parent blob is implicit directory and in case the blob deleted was the
     * only blob in the directory, it will render path as non-existing. To prevent
     * that happening, it is needed to  create marker-based directory on the
     * parentPath. This is inspired from WASB implementation.
     */
```